src/services/fields.ts

```diff
@@ -142,6 +143,58 @@ export function getDetectedFieldsJsonPathField(detectedFieldsFrame?: DataFrame)
   const pathField: Field<string[]> | undefined = detectedFieldsFrame?.fields[4];
   return pathField;
 }
+export function getEstimatedCardinality(labelName: string, detectedFieldsFrame: DataFrame | undefined) {
+  const namesField = getDetectedFieldsNamesField(detectedFieldsFrame);
+  const cardinalityField = getDetectedFieldsCardinalityField(detectedFieldsFrame);
+  const index = namesField?.values.findIndex((name) => name === labelName);
+  if (index !== undefined && index !== -1) {
+    return cardinalityField?.values?.[index];
+  }
+
+  return undefined;
+}
+
+export interface SparsityCalculation {
+  cardinality: number | undefined;
+  description: string | undefined;
+  sparsity: number | undefined;
+}
+
+export function calculateSparsity(sceneRef: SceneObject, labelName: string): SparsityCalculation {
```

Comment:
Too me this function is calculating label prevalence or label coverage (names are hard), sparsity usually refers to zero or null datasets.